### PR TITLE
[test.py] Add uniqueness to the test name

### PR DESCRIPTION
**Pull request by xtrey**
In CI test always executed with option --repeat=3 that leads to generate 3 test results with the same name. Junit plugin in CI cannot distinguish correctly the difference between these results. In case when we have two passes and one fail, the link to test result will sometimes be redirected to the incorrect one because the test name is the same. To fix this ReportPlugin added that will be responsible to modify the test case name during junit report generation adding to the test name mode and run id.

Fixes: https://github.com/scylladb/scylladb/issues/17851

Fixes: https://github.com/scylladb/scylladb/issues/15973

### CODE REVIEW

**Comment by avikivity** [comment]
Please use Fixes rather than Closes, we reserve Closes to closing pull requests by the promotion bot.

**Comment by avikivity** [comment]
Otherwise, looks reasonable.

File: test/alternator/conftest.py [hunk 1]
```diff
@@ -13,6 +13,10 @@
 import boto3
 import requests
 import re
+
+import sys
+sys.path.insert(1, sys.path[0] + '/../..')
```
Comment (kbr-scylla) [inline test/alternator/conftest.py:18]: This is ugly and breaks linters / type checkers (mypy in particular).

Why do you do it? I don't think it's needed.

File: test/cql-pytest/conftest.py [hunk 2]
```diff
@@ -21,6 +21,9 @@
 import time
 import random
 
+import sys
+sys.path.insert(1, sys.path[0] + '/../..')
```
Comment (kbr-scylla) [inline test/cql-pytest/conftest.py:25]: See above

File: test/alternator/conftest.py (hunk 1 above)
Comment (kbr-scylla) [reply test/alternator/conftest.py:18]: I put in work to remove these in the past: ff4430d8ea1f6dfc38c43add38822fdefd773f1a

and now you're bringing them back :(

File: test/nodetool/conftest.py [hunk 3]
```diff
@@ -15,6 +15,8 @@
 from typing import NamedTuple
 
 import rest_api_mock
+sys.path.insert(1, sys.path[0] + '/../..')
```
Comment (kbr-scylla) [inline test/nodetool/conftest.py:18]: See above

File: test/rest_api/conftest.py [hunk 4]
```diff
@@ -18,6 +18,9 @@
 from cassandra.cluster import Cluster, ConsistencyLevel, ExecutionProfile, EXEC_PROFILE_DEFAULT
 from cassandra.policies import RoundRobinPolicy
 
+sys.path.insert(1, sys.path[0] + '/../..')
```
Comment (kbr-scylla) [inline test/rest_api/conftest.py:21]: See above

**Review by kbr-scylla — CHANGES_REQUESTED** [review]

File: test/alternator/conftest.py (hunk 1 above)
Comment (kbr-scylla) [reply test/alternator/conftest.py:18]: Perhaps some subdirectories are missing __init__.py files?

Comment (xtrey) [reply test/alternator/conftest.py:18]: Fully agree that modifying the sys path is ugly, and inconvenient, but making it a package will require a lot of refactorings, and it's not a matter of this PR.

File: test/cql-pytest/conftest.py (hunk 2 above)
Comment (xtrey) [reply test/cql-pytest/conftest.py:25]: To make it a package the directory should be renamed to cql_pytest, python have a regex that not allow directories with '-' to be a package. But such proposal was declined by the owner https://github.com/scylladb/scylladb/pull/11395

File: test/nodetool/conftest.py (hunk 3 above)
Comment (xtrey) [reply test/nodetool/conftest.py:18]: Same here, it's not an easy fix, it will require some refactoring.

File: test/rest_api/conftest.py (hunk 4 above)
Comment (xtrey) [reply test/rest_api/conftest.py:21]: Unfortunately, here the situation is the same.

**Comment by xtrey** [comment]
Version 2:
moved scylla-gdb to scylla_gdb to make it a package to fix failed tests

**Comment by xtrey** [comment]
Version 3:
- commit message changed

**Review by kbr-scylla — APPROVED** [review]

**Comment by nyh** [comment]
Sadly, this patch broke test/cql-pytest/run. It now demands a "--mode" option, which it mustn't.
This is so sad. I keep asking you NOT TO BREAK test/{alternator,cql-pytest}/conftest.py, or at least let me review it.
Now I need to unbreak it.

**Pull request merged**